edu.gemini.lch.model/src/main/java/edu/gemini/lch/model/Interval.scala

```diff
@@ -0,0 +1,28 @@
+package edu.gemini.lch.model
+
+import java.time.Instant
+
+/**
+ * An interval is a period of time between two instants.
+ * Includes start, excludes end.
+ * Requirement: end >= start.
+ * They have fixed ms duration, i.e. end - start and are not comparable.
+ * To compare intervals, you should compare their lengths, i.e. Durations.
+ */
+case class Interval(start: Instant, end: Instant) {
+  def overlaps(other: Interval): Boolean = {
+    val thisStart = start.toEpochMilli
+    val thisEnd   = end.toEpochMilli
+
+    // If null, assume now and check if now falls in the represented interval.
+    if (other == null) {
+      val now: Instant = Instant.now()
+      thisStart < now.toEpochMilli && now.toEpochMilli < thisEnd
```

Comment:
This is unexpected and obviously not referentially transparent.  I bet, though, that it is maintaining some "feature" of Joda Time?  I would say it should blow up with a NPE if you hand it a `null`.

Reply:
There were warnings about `null` being treated differently between joda and Java 8 time. We'll invariably have to sift through a few NPEs at some point. It doesn't seem remotely logical to me to assume that a null `Interval` is overlapped if the the "overlapping" interval contains now?

Reply:
I'll give it an NPE for now to resolve this conversation and make this more logical.

Reply:
Or perhaps I shouldn't even generate the NPE and let the compiler handle that?